### PR TITLE
Image/Attachment field

### PR DESCRIPTION
**Pull request by mattheu**
I've had a few requests for an image field that doesn't require the whole of fieldmanager for those who don't use it.

@todo Should handle restricting library by file type

See #106

### CODE REVIEW

File: css/sass/shortcode-ui.scss [hunk 1]
```diff
@@ -128,3 +132,68 @@
 
 }
 
+
+.edit-shortcode-form .shortcake-attachment-preview {
+
+	width: 150px !important;
+	height: 150px !important;
```
Comment (danielbachhuber) [inline css/sass/shortcode-ui.scss]: Why do we need these `!important`?

File: dev.php [hunk 2]
```diff
@@ -44,8 +49,8 @@
 			// Display label. String. Required.
 			'label' => 'Shortcake Dev',
 
-			// Icon/image for shortcode. Optional. src or dashicons-$icon. Defaults to carrot.
-			'listItemImage' => 'dashicons-editor-quote',
+			// Icon/attachment for shortcode. Optional. src or dashicons-$icon. Defaults to carrot.
+			'listItemAttachment' => 'dashicons-editor-quote',
```
Comment (danielbachhuber) [inline dev.php]: Why is this example changing?

File: inc/fields/class-field-attachment.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+	 */
+	public function action_print_media_templates() {
+
+		?>
+
+		<script type="text/html" id="tmpl-fusion-shortcake-field-attachment">
+			<div class="field-block">
+				<label for="{{ data.attr }}">{{ data.label }}</label>
+				<div class="shortcake-attachment-preview attachment-preview attachment">
+					<button id="{{ data.attr }}" class="button button-small add">Select Attachment</button>
```
Comment (danielbachhuber) [inline inc/fields/class-field-attachment.php]: String should be translatable.

File: inc/fields/class-field-attachment.php [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+	/**
+	 * Initialize Post Fields.
+	 *
+	 * Find all registered shortcode attributes that use the post field and
+	 * store an array with all their args for future reference.
+	 *
+	 * @return null
+	 */
+	public function action_init_post_field() {
+	}
```
Comment (danielbachhuber) [inline inc/fields/class-field-attachment.php]: Do we need this method?

File: shortcode-ui.php [hunk 5]
```diff
@@ -21,11 +21,14 @@
 
 require_once dirname( __FILE__ ) . '/inc/class-shortcode-ui.php';
 require_once dirname( __FILE__ ) . '/inc/fields/class-shortcode-ui-fields.php';
+require_once dirname( __FILE__ ) . '/inc/fields/class-field-attachment.php';
+require_once dirname( __FILE__ ) . '/dev.php';
```
Comment (danielbachhuber) [inline shortcode-ui.php]: Please remove this `/dev.php`

File: inc/fields/class-field-attachment.php [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+			self::$instance->setup_actions();
+		}
+		return self::$instance;
+	}
+
+	private function setup_actions() {
+
+		add_filter( 'shortcode_ui_fields', array( $this, 'filter_shortcode_ui_fields' ) );
+		add_action( 'admin_enqueue_scripts', array( $this, 'action_admin_enqueue_scripts' ), 100 );
+		add_action( 'print_media_templates', array( $this, 'action_print_media_templates' ) );
```
Comment (danielbachhuber) [inline inc/fields/class-field-attachment.php]: Can we make sure these are only loaded when Shortcode UI is loaded?

**Comment by bfintal** [comment]
+1 to this

**Comment by roborourke** [comment]
+1, the fieldmanager one seems to be broken at the moment anyway

**Comment by bfintal** [comment]
Tested, and works great :+1:

**Comment by roborourke** [comment]
Nice one Matt. Works really well, cheers for adding in the library type feature too

**Comment by mattheu** [comment]
Made a few updates - I think this is good to go.

**Comment by danielbachhuber** [comment]
@mattheu We can probably ditch Fieldmanager media field then, no?

**Comment by mattheu** [comment]
Yes - aside from backwards compatability with anyone using it already.

**Comment by danielbachhuber** [comment]
Looks pretty solid. Only one minor issue #155

**Pull request merged**